### ISSUE TITLE
Twig: refactor Icon component 

### ISSUE THREAD

**Issue by justintemps** [international-labour-organization/designsystem#692]
## Description

Right now the icon component in Twig looks like this:

```
<img class="{{prefix}}--icon" src="{{'/images/'~icon~'.svg'}}" alt="{{icon}}" role="presentation">
```

This is sort of a naive implementation that assumes there's some top-level `/images` directory that has all of the icons in the Design System. That's true in the Twig package itself, and so it works in Storybook, but it's not the case in most Drupal Websites.

We need to refactor this component so that it's either using CSS to render the icons as background-images from `@ilo-org/icons/build/icons.scss` or we need to find the standard Twig way of telling the server that the path to the icons is actually in `/@ilo-org/icons/svg`

## Example 

See the mobile version of this page: https://iloorgstg.prod.acquia-sites.com/node/56

![image](https://github.com/international-labour-organization/designsystem/assets/12401179/bb629301-e27d-4225-bdb5-e1a2a426512a)

**Comment by justintemps** [comment]
Let's explore the option of having a simple script that a Drupal developer can run in order to place all of the image files from the token package into a place where the Token component will be expecting to find them. We may also need to add a path arg to the Token component so that the Drupal development can keep icons somewhere other than `/images`.

**Comment by GGKapanadze** [comment]
## Introduction

The new `icon` component should be done without any directory generation or "run an extra script" step for twig implementation. In the current context, we have 2 main ways 
- web fonts
- SVG sprite

tl;dr web fonts will 
- require complex build steps for fonts
- add additional effort for colorized icons
- make debugging more complex

IMO, SVG sprite is the good for our scenario. The main idea of Sprite is that we define one `svg` element and using `symbol`-s we can add multiple items inside of `defs`, after that we can reference any symbol with `use` keyword and `id` on each `symbol`.  

https://github.com/international-labour-organization/designsystem/assets/36326203/6acb4b61-b164-43c7-a91f-292345d9cc03


`svg` sprite is almost perfect in our case but it will require us to have some idea how are we doing the string concatenation for the sprite, aka how to unionize all icons into one big `svg`.  While I was playing with the concatenation problem with `@ilo-org/icon` build tools, I noticed that we have an `umd` export that contains all icons with all sizing bundled in one file as plain Javascript objects. It looks exactly like the sprite I mentioned above but with objects, so instead of `use` keyword we will utilize `JS` to create SVG elements from the exported format

<img width="883" alt="image" src="https://github.com/international-labour-organization/designsystem/assets/36326203/594718f0-992b-47da-9f65-b6193ba01bd0">


## Implementation

For the implementation, we will use a name prop as an icon name like this
```twig
<svg 
 class="prefix: ilo--icon" 
 data-prefix="prefix: ilo"
 data-name="{{name}}" {# name of the icon #}
 data-loadcomponent="Icon">
</svg>
```
after that will map the given name to `umd` export from `@ilo-org/icons` and finally we will build the svg.

https://github.com/international-labour-organization/designsystem/assets/36326203/4fdbab76-fedd-443d-9284-496428afddd8

Quick prototype powering the demo above
```js
import * as Icons from "@ilo-org/icons";

export default class Icon {
  constructor(element) {
    this.element = element;
    this.init();
  }

  init() {
    const config = Icons[this.element.dataset.name];
    const element = document.createElementNS(
      "http://www.w3.org/2000/svg",
      "svg"
    );

    for (const [key, value] of Object.entries(config.attrs)) {
      element.setAttribute(key, value);
    }

    for (const item of config.content) {
      const child = document.createElementNS(
        "http://www.w3.org/2000/path",
        item.elem
      );
      for (const [key, value] of Object.entries(item.attrs).reverse()) {
        child.setAttribute(key, value);
      }
      element.appendChild(child);
    }

    this.element.outerHTML = element.outerHTML;
    return this;
  }
}

```

With the approach we 
- Will utilize native `@ilo-org/icons` library
- Prevent any script running or directory creation
- Avoid extra build step for 1.0

@bashlk @justintemps

**Comment by bashlk** [comment]
@GGKapanadze How will this work with twig + Drupal? 

My previous understanding of the sprite idea is that we have a single file with all the icons "imported globally" and the icon component would look up individual icons within the sprite. In this case, would this be done by the bundler with the wildcard import of the icon package?

**Comment by GGKapanadze** [comment]
Sure but the Drupal repo already has `@ilo-org/icons` as a [dependency](https://github.com/international-labour-organization/ilo-org-drupal/blob/main/docroot/themes/custom/ilo/package.json#L61) and we will add sprite on top of it.

If we are going to generate the sprite, it should be part of `icons` package, which will export a `svg` string, which will be consumed by the JS bundle and attached to the DOM
